### PR TITLE
FunctionActivator for SingleStore TVFs

### PR DESCRIPTION
**Pull request by satendra-sahu**
#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-xts-memsqlFunction/legend-engine-xt-memsqlFunction-api/src/main/java/org/finos/legend/engine/language/memsql/deployment/MemSqlFunctionArtifactGenerationExtension.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.dsl.generation.extension.Artifact;
+import org.finos.legend.engine.language.pure.dsl.generation.extension.ArtifactGenerationExtension;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContextData;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.PackageableElement;
+
+import java.util.List;
+
+public class MemSqlFunctionArtifactGenerationExtension implements ArtifactGenerationExtension
```
Comment (Yasirmod17) [inline legend-engine-xts-memsqlFunction/legend-engine-xt-memsqlFunction-api/src/main/java/org/finos/legend/engine/language/memsql/deployment/MemSqlFunctionArtifactGenerationExtension.java:25]: can you call this MemSQL..

File: legend-engine-xts-memsqlFunction/legend-engine-xt-memsqlFunction-api/src/main/java/org/finos/legend/engine/language/memsql/deployment/MemSqlFunctionArtifactGenerationExtension.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Goldman Sachs
```
Comment (Yasirmod17) [inline legend-engine-xts-memsqlFunction/legend-engine-xt-memsqlFunction-api/src/main/java/org/finos/legend/engine/language/memsql/deployment/MemSqlFunctionArtifactGenerationExtension.java:1]: Any reason why we have this just empty? should we remove it

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-memsql/legend-engine-xt-relationalStore-memsql-execution/src/main/java/org/finos/legend/engine/authentication/flows/MemSQLStaticWithUserPasswordFlow.java [hunk 3]
```diff
@@ -15,20 +15,20 @@
...
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.DatabaseType;
 import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.authentication.UserNamePasswordAuthenticationStrategy;
-import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.specification.StaticDatasourceSpecification;
 import org.finos.legend.engine.shared.core.identity.Credential;
 import org.finos.legend.engine.shared.core.identity.Identity;
 import org.finos.legend.engine.shared.core.identity.credential.PlaintextUserPasswordCredential;
 import org.finos.legend.engine.shared.core.vault.Vault;
 
-public class MemSQLStaticWithUserPasswordFlow implements DatabaseAuthenticationFlow<StaticDatasourceSpecification, UserNamePasswordAuthenticationStrategy>
+public class MemSQLStaticWithUserPasswordFlow implements DatabaseAuthenticationFlow<MemSqlDatasourceSpecification, UserNamePasswordAuthenticationStrategy>
```
Comment (Yasirmod17) [inline legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-memsql/legend-engine-xt-relationalStore-memsql-execution/src/main/java/org/finos/legend/engine/authentication/flows/MemSQLStaticWithUserPasswordFlow.java:26]: why do we need to change this?

**Review by Yasirmod17 — CHANGES_REQUESTED** [review]
Lets sync on this

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-memsql/legend-engine-xt-relationalStore-memsql-execution/src/main/java/org/finos/legend/engine/authentication/flows/MemSQLStaticWithUserPasswordFlow.java (hunk 3 above)
Comment (satendra-sahu) [reply legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-memsql/legend-engine-xt-relationalStore-memsql-execution/src/main/java/org/finos/legend/engine/authentication/flows/MemSQLStaticWithUserPasswordFlow.java:26]: Added a new DatasourceSpecification for MemSQL that has SSL attribute

File: legend-engine-xts-memsqlFunction/legend-engine-xt-memsqlFunction-api/src/main/java/org/finos/legend/engine/language/memsql/deployment/MemSqlFunctionArtifactGenerationExtension.java (hunk 2 above)
Comment (satendra-sahu) [reply legend-engine-xts-memsqlFunction/legend-engine-xt-memsqlFunction-api/src/main/java/org/finos/legend/engine/language/memsql/deployment/MemSqlFunctionArtifactGenerationExtension.java:1]: It was a duplicate class hence removed.

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-memsql/legend-engine-xt-relationalStore-memsql-pure/src/main/resources/core_relational_memsql.definition.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "core_relational_memsql",
-  "pattern": "(meta::relational::functions::sqlQueryToString::memsql|meta::relational::tests::sqlQueryToString::memsql|meta::relational::memsql::tests)(::.*)?",
+  "pattern": "(meta::relational::functions::sqlQueryToString::memsql|meta::pure::alloy::connections|meta::protocols::pure|meta::relational::tests::sqlQueryToString::memsql|meta::relational::memsql::tests)(::.*)?",
```
Comment (Yasirmod17) [inline legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-memsql/legend-engine-xt-relationalStore-memsql-pure/src/main/resources/core_relational_memsql.definition.json:3]: There should be no reference to Alloy in here. Lets look to refactor whatever needs this package

File: legend-engine-xts-text/legend-engine-xt-text-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/TextProtocolExtension.java [hunk 5]
```diff
@@ -47,6 +47,6 @@ public List<Function0<List<ProtocolSubTypeInfo<?>>>> getExtraProtocolSubTypeInfo
     @Override
     public Map<Class<? extends PackageableElement>, String> getExtraProtocolToClassifierPathMap()
     {
-        return Maps.mutable.with(Text.class, "meta::pure::metamodel::text::Text");
+        return Maps.mutable.with(Text.class, "meta::pure::metamodel::text: :Text");
```
Comment (Yasirmod17) [inline legend-engine-xts-text/legend-engine-xt-text-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/TextProtocolExtension.java:50]: remove please

File: legend-engine-xts-memsqlFunction/legend-engine-xt-memsqlFunction-api/src/main/java/org/finos/legend/engine/language/memsql/deployment/MemSqlFunctionDeploymentManager.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+
+/**
+ * These deployment functions assume that the artifact has already been validated.
+ */
+public class MemSqlFunctionDeploymentManager implements DeploymentManager<MemSqlFunctionArtifact, MemSqlFunctionDeploymentResult, MemSqlFunctionDeploymentConfiguration>
+{
```
Comment (Yasirmod17) [inline legend-engine-xts-memsqlFunction/legend-engine-xt-memsqlFunction-api/src/main/java/org/finos/legend/engine/language/memsql/deployment/MemSqlFunctionDeploymentManager.java:45]: add ovveride method for interface

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-memsql/legend-engine-xt-relationalStore-memsql-pure/src/main/resources/core_relational_memsql/relational/protocols/pure/v1_20_0/extension/metamodel_conection.pure [hunk 7]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2023 Goldman Sachs
+//
```
Comment (Yasirmod17) [inline legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-memsql/legend-engine-xt-relationalStore-memsql-pure/src/main/resources/core_relational_memsql/relational/protocols/pure/v1_20_0/extension/metamodel_conection.pure:2]: can remove all the protocol files. only vX_X_X

**Pull request closed without merging**